.docker/ci/Dockerfile

```diff
@@ -1,50 +1,82 @@
-RUN \
-    mkdir src && \
-    cd src && \
-    #
-    # Download moveit source, so that we can get necessary dependencies
-    wstool init --shallow . https://raw.githubusercontent.com/ros-planning/moveit/${ROS_DISTRO}-devel/moveit.rosinstall && \
-    #
-    # Update apt package list as previous containers clear the cache
-    apt-get -qq update && \
-    apt-get -qq dist-upgrade && \
-    #
-    # Install some base dependencies
-    apt-get -qq install -y \
-        # Some source builds require a package.xml be downloaded via wget from an external location
-        wget \
-        # Required for rosdep command
-        sudo \
-        # Preferred build tools
-        python-catkin-tools \
-        clang clang-format-3.9 clang-tidy clang-tools \
-        ccache && \
-    #
-    # Download all dependencies of MoveIt!
-    rosdep update && \
-    rosdep install -y --from-paths . --ignore-src --rosdistro ${ROS_DISTRO} --as-root=apt:false && \
-    #
-    # Remove the source code from this container
-    # TODO: in the future we may want to keep this here for further optimization of later containers
-    cd .. && \
-    rm -rf src/ && \
-    #
-    # Clear apt-cache to reduce image size
-    rm -rf /var/lib/apt/lists/*
-
-# Continous Integration Setting
-ENV IN_DOCKER 1
+# This dockerfile expects proxies to be set via --build-arg if needed
+# It also expects to be contained in the /moveit2 root folder for file copy
+# Example build command:
+# export http_proxy=http://my.proxy.com:80
+# export CMAKE_BUILD_TYPE=Debug
+# docker build -t nav2:latest --build-arg http_proxy --build-arg CMAKE_BUILD_TYPE ./
+FROM osrf/ros2:nightly
+
+# setup keys
+ARG http_proxy
+RUN if [ "$http_proxy" != "" ]; \
+    then \
+      apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 \
+      --keyserver-options http-proxy=$http_proxy \
+      --recv-keys 421C365BD9FF1F717815A3895523BAEEB01FA116; \
+    fi
+
+# install ROS2 dependencies
+RUN apt-get update && apt-get install -q -y \
+      build-essential \
+      cmake \
+      git \
+      python3-colcon-common-extensions \
+      python3-vcstool \
+      wget \
+    && rm -rf /var/lib/apt/lists/*
+
+# copy ros package repo
+ENV MOVEIT2_WS /opt/moveit2_ws
+RUN mkdir -p $MOVEIT2_WS/src
+WORKDIR $MOVEIT2_WS/src
+COPY ./ moveit2/
+
+# clone dependency package repos
+ENV ROS_WS /opt/ros_ws
+RUN mkdir -p $ROS_WS/src
+WORKDIR $ROS_WS
+RUN vcs import src < $MOVEIT2_WS/src/moveit2/tools/ros2_dependencies.repos
+
+# install dependency package dependencies
+RUN . /opt/ros/$ROS_DISTRO/setup.sh && \
+    apt-get update && \
+    rosdep install -q -y \
+      --from-paths \
+        src \
+      --ignore-src \
+    && rm -rf /var/lib/apt/lists/*
+
+# build dependency package source
+ARG CMAKE_BUILD_TYPE=Release
+
+RUN . /opt/ros/$ROS_DISTRO/setup.sh && \
+    colcon build \
+      --symlink-install \
+      --cmake-args \
+        -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE
+
+# install moveit2 package dependencies
+WORKDIR $MOVEIT2_WS
```

Comment:
isn't this redundant

Reply:
Not quite, this is changing the pwd to the second workspace overlay to build moveit2. The first, `ROS2_WS` was  reserved for any upstream dependencies this project necessitates be built from source.

Reply:
I se